components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/policy/finder/CarbonPolicyFinder.java

```diff
@@ -223,6 +223,7 @@ public PolicyFinderResult findPolicy(EvaluationCtx context) {
             init(this.finder);
             policyReferenceCache.clear();
             EntitlementEngine.getInstance().clearDecisionCache();
+            EntitlementEngine.getInstance().getPolicyCache().resetCacheInvalidateState();
```

Comment:
What is the guarantee that EntitlementEngine.getInstance().getPolicyCache() is not null always? 
Better option is to add a method to EntitlementEngine.class
public void clearPolicyCacheForCurrentTenent() {
 if(getPolicyCache() != null) {
getPolicyCache(). resetCacheInvalidateState();
}